mautic/mautic
# Social monitoring and tweet contacts via campaign

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y 
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  #1742

## Description

This PR does several things:

1. It brings the features of mPower Social to the community courtesy of mautic.com! This provides the ability to create contacts from Twitter mentions or handles and to tweet leads as part of a campaign (your responsibility to be compliant with Twitter's automation rules https://support.twitter.com/articles/76915#Tweets :-) 

## Steps to test this PR

New Features:
1. Apply the PR and clear the cache
2. Run `doctrine:migrations:migrate`
3. Ensure Twitter is authorized in Plugins
4. Go to Channels -> Social Monitoring and create a monitor
5. Run the command `mautic:social:monitor`
6. Click on the monitor in Social Monitoring and look at the leads tab. Click on a lead and you should see the social profile field filled in for Twitter
7. Create a basic campaign with a tweet contact action (**WARNING: be sure you are not tweeting every one you just pulled from a monitor :-D** )
8. Trigger the campaign and the contact should get a tweet
